core/src/main/java/nl/nn/adapterframework/filesystem/MsalClientAdapter.java

```diff
@@ -0,0 +1,222 @@
+
+public class MsalClientAdapter extends HttpSenderBase implements IHttpClient {
+	private static final String METHOD_SESSION_KEY = "HTTP_METHOD";
+	private static final String REQUEST_HEADERS_SESSION_KEY = "HTTP_REQUEST_HEADERS";
+	private static final String RESPONSE_HEADERS_SESSION_KEY = "HTTP_RESPONSE_HEADERS";
+	private static final String URL_SESSION_KEY = "URL";
+	private static final String STATUS_CODE_SESSION_KEY = "HTTP_STATUSCODE";
+
+	public MsalClientAdapter() {
+		setName("MSAL");
+
+		Parameter urlParameter = new Parameter();
+		urlParameter.setName("url");
+		urlParameter.setSessionKey(URL_SESSION_KEY);
+		addParameter(urlParameter);
+
+		setResultStatusCodeSessionKey(STATUS_CODE_SESSION_KEY);
+	}
+
+	@Override
+	public IHttpResponse send(HttpRequest httpRequest) throws Exception {
+		PipeLineSession session = prepareSession(httpRequest);
+		Message request = new Message(httpRequest.body());
+
+		try {
+			Message response = sendMessage(request, session);
+			return translateResponse(response, session);
+		} catch (Exception e) {
+			log.error("An exception occurred whilst connecting with MSAL HTTPS call to [" + httpRequest.url().toString() + "]", e);
+			throw e;
+		} finally {
+			session.close();
+		}
+	}
+
+	private PipeLineSession prepareSession(HttpRequest httpRequest) {
+		PipeLineSession session = new PipeLineSession();
+		session.put(URL_SESSION_KEY, httpRequest.url().toString());
+		log.debug("Put request URL [{}] in session under key [{}]", ()->httpRequest.url().toString(), ()->URL_SESSION_KEY);
+
+		session.put(METHOD_SESSION_KEY, httpRequest.httpMethod().name());
+		log.debug("Put http method [{}] in session under key [{}]", ()->httpRequest.httpMethod().name(), ()->METHOD_SESSION_KEY);
+
+		session.put(REQUEST_HEADERS_SESSION_KEY, httpRequest.headers());
+		if(log.isDebugEnabled()) log.debug("Put http headers [{}] in session under key [{}]", ()->httpRequest.headers(), ()->REQUEST_HEADERS_SESSION_KEY);
+
+		return session;
+	}
+
+	@Override
+	protected HttpRequestBase getMethod(URI uri, Message message, ParameterValueList parameters, PipeLineSession session) throws SenderException {
+		HttpMethod httpMethod = EnumUtils.parse(HttpMethod.class, (String) session.get(METHOD_SESSION_KEY));
+		Map<String, String> headers = (Map<String, String>) session.get(REQUEST_HEADERS_SESSION_KEY);
+
+		if(uri == null) {
+			throw new SenderException("No URI to connect to!");
+		}
+
+		boolean queryParametersAppended = false;
+		StringBuffer relativePath = new StringBuffer(uri.getRawPath());
+		if(!StringUtils.isEmpty(uri.getQuery())) {
+			relativePath.append("?" + uri.getQuery());
+			queryParametersAppended = true;
+		}
+		try {
+			switch (httpMethod) {
+			case GET:
+				if(parameters != null) {
+					queryParametersAppended = appendParameters(queryParametersAppended, relativePath, parameters);
+					if(log.isDebugEnabled())
+						log.debug(getLogPrefix() + "path after appending of parameters [" + relativePath + "]");
+				}
```

Comment:
er zijn geen parameters, dit is toch niet nodig?

Reply:
eh, wellicht, dit is een copy-paste geweest van http sender :)